### PR TITLE
Feat: Open New Tab Settings

### PR DESCRIPTION
**Pull request by sshanzel**
Related [PR](https://github.com/dailydotdev/daily-apps/pull/135).

Hi Team,

As I was unable to push changes on OP's forked repo, I had to create a separate PR for the feature request.

Opening New Tab when hitting the links should be based on the user's preference and the default value should be `true`.

Kindly check the changes and please just send feedback with regards to this PR if there's any issue or concern so I can further make the necessary changes.

Thank you!

Best Regards,
Lee

### CODE REVIEW

**Comment by sshanzel** [comment]
Oops. Something failed. Will work on it in a few :grin:

**Comment by sshanzel** [comment]
Hi @idoshamun, can you please help me identify where could this failed test case be coming from? I

File: packages/components/src/common/postMixin.js [hunk 1]
```diff
@@ -69,7 +73,7 @@ export default {
     },
 
     onLinkClick() {
-      this.$emit('click', this.post);
+      this.$emit('click');
```
Comment (idoshamun) [inline packages/components/src/common/postMixin.js:76]: This line is the reason for the failed builds.
Make sure to accept my suggestion to fix the issue.

```suggestion
      this.$emit('click', this.post);
```

File: packages/components/src/common/postMixin.js [hunk 2]
```diff
@@ -106,6 +110,10 @@ export default {
   },
 
   computed: {
+    shouldOpenNewTab() {
+      return this.openNewTab ? "_blank" : "_self";
```
Comment (idoshamun) [inline packages/components/src/common/postMixin.js:114]: Please make sure to run `yarn lint --fix`, e.g we do not use double quotes.
I need to fix the CI to throw error on linting issues 😅

File: packages/extension/src/components/DaFeed.vue [hunk 3]
```diff
@@ -14,7 +14,7 @@
                         @upvote="onUpvote" @comment="onComment"
                         :menu-opened="selectedPostId === item.id"
                         :bookmarks-menu-opened="bookmarkPostId === item.id"
-                        :selected="focusedPost === item"
+                        :selected="focusedPost === item" :openNewTab="openNewTab"
```
Comment (idoshamun) [inline packages/extension/src/components/DaFeed.vue:17]:
```suggestion
                        :selected="focusedPost === item" :open-new-tab="openNewTab"
```

File: packages/extension/src/components/DaFeed.vue [hunk 4]
```diff
@@ -33,7 +33,7 @@
                       @upvote="onUpvote" @comment="onComment"
                       :menu-opened="selectedPostId === item.id"
                       :bookmarks-menu-opened="bookmarkPostId === item.id"
-                      :selected="focusedPost === item"
+                      :selected="focusedPost === item" :openNewTab="openNewTab"
```
Comment (idoshamun) [inline packages/extension/src/components/DaFeed.vue:36]:
```suggestion
                      :selected="focusedPost === item" :open-new-tab="openNewTab"
```

File: packages/components/src/common/postMixin.js (hunk 1 above)
Comment (sshanzel) [reply packages/components/src/common/postMixin.js:76]: Ohhh, so this is why. Thank you!

File: packages/components/src/common/postMixin.js (hunk 2 above)
Comment (sshanzel) [reply packages/components/src/common/postMixin.js:114]: Oh sorry, I forgot. Lol.

Comment (sshanzel) [reply packages/components/src/common/postMixin.js]: Hi bro, I tried running the command and it was not changed, though I already modified it manually :grin:

You might get interested on `Prettier`. It automatically formats the code on save (default) based on configuration also and has a nice VS Code extension too!

**Comment by sshanzel** [comment]
Thank you for the careful review! I've now made the necessary changes, the build has passed the tests also, please let me know if there's any issue or concern so I can push updates immediately.

**Comment by sshanzel** [comment]
Thanks a lot man! 🍻

**Pull request merged**